### ISSUE TITLE
About perfect result

### ISSUE THREAD

**Issue by sunboyan1024** [vicoslab/mixed-segdec-net-comind2021#18]
Thanks for this great work. When I use a GTX2080TI 11G with an epoch of 50 rounds, the result of the SDD2 data set cannot be reproduced, which is 95.4%. My best result was only 92.6%. Can you give me some suggestions？

**Comment by JakobBozic** [comment]
Hi, I have also noticed that results seem to degrade with the modified version of the distance transform for the segmentation loss mask that was included after the release. If you are using the most recent version of the code, try reverting to the first commit from this repository (ffd947468af71f566bdbfe4d99ec1cc899db55dc). Additionally in end2end.py in line 47, put the             torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
outside of the if clause so that it always applies. 

Apparently, the modified version of the distance transform gives a bit worse results, I will look further into it and remove it if necessary. Have you also tried running the experiment with fewer segmented positive samples?

**Comment by sunboyan1024** [comment]
Yes.Now when I tried to train KSDD when N=0,the result is only  40.6%(AP).Other results is right.
The code is  "train_KSDD $SAVE_IMAGES N_0    $RESULTS_PATH 7 33 0  50 1 0.01 1 True  2 1 False True  True  "${GPUS[@]}""

**Comment by JakobBozic** [comment]
Hi, 
there was a mistake in EXPERIMENTS_COMIND.sh, for N=0, DELTA_CLS_LOSS should be 1 and LEARNING_RATE 0.01. I have fixed it, the correct code is "train_KSDD $SAVE_IMAGES N_0    $RESULTS_PATH 7 33 0  50 **0.01 1** 1 True  2 1 False True  True  "${GPUS[@]}"".
